GWFish/modules/waveforms.py

```diff
@@ -10,29 +11,43 @@
     from lal import CreateREAL8Vector
 except ModuleNotFoundError as err:
     uselal = err
-    print('LAL package is not installed. Only GWFish waveforms available.')
+    logging.warning('LAL package is not installed. Only GWFish waveforms available.')
 
 import GWFish.modules.constants as cst
 import GWFish.modules.auxiliary as aux
+import GWFish.modules.fft as fft
 
-def hphc_amplitudes(waveform, parameters, frequencyvector, plot=None):
+def hphc_amplitudes(waveform, parameters, frequencyvector, time_domain=False, plot=None):
     parameters = parameters.copy()
 
     if waveform == 'gwfish_TaylorF2':
         hphc = TaylorF2(parameters, frequencyvector, plot=plot)
     elif waveform == 'gwfish_IMRPhenomD':
         hphc = IMRPhenomD(parameters, frequencyvector, plot=plot)
     elif waveform[0:7] == 'lalsim_':
-        hphc = lal_caller(waveform[7:], frequencyvector, **parameters)
+        if time_domain:
+            data_params = {'frequencyvector': frequencyvector}
+            waveform_obj = LALTD_Waveform(waveform[7:], parameters, data_params)
+            hphc = waveform_obj()
+        else:
+            data_params = {'frequencyvector': frequencyvector}
+            waveform_obj = LALFD_Waveform(waveform[7:], parameters, data_params)
+            hphc = waveform_obj()
     else:
-        print(str(waveform) + ' is not a valid waveform.')
-        print('Valid options are gwfish_TaylorF2, gwfish_IMRPhenomD, lalsim_XXX.')
-
-    t_of_f = t_of_f_PN(parameters, frequencyvector)
+        waveform_error = '{} is not a valid waveform. '.format(str(waveform)) + \
+                         'Valid options are gwfish_TaylorF2, gwfish_IMRPhenomD, lalsim_XXX.'
+        raise ValueError(waveform_error)
+
+    if time_domain:
+        # Here it is not really t_of_f, it is just a time vector
+        t_of_f = np.arange(0,hphc.shape[0]*waveform_obj.delta_t,waveform_obj.delta_t)
```

Comment:
This time vector starts from 0, but the user can pass `geocent_time`, the GPS time of merger: this will then be ignored at the moment if the waveform is in the time domain.

Here there should be either a `+parameters['geocent_time']` (and the time array being negative approaching zero), or starting from the frequencyvector (which you have anyway as a property of the `Waveform` object) you can use the `t_of_f_PN` function (less precise, more consistent with the FD waveforms.

Reply:
In fact, I think the original `t_of_f` function is works well here even for time-domain waveforms too. So, I removed this part, only keeping the original code.

Reply:
OK, so if I understand correctly now the `t_of_f` function defined in the base `Waveform` object will be called for all waveforms, be they from TD or FD directly. Seems sensible!